### PR TITLE
main/cmake: update to 3.27.3

### PR DESCRIPTION
**Pull request by nekopsykose**
also uses ninja for cmake's self-bootstrap to make it more faster since it's free

### CODE REVIEW

**Comment by nekopsykose** [comment]
the sigill with cfi is unrelated to the upgrade and also just fails on 3.27.1. unsetting it works (and is fine, since this is neither a longrun service nor super meaningful for hardening), but perhaps you want to explore that

**Comment by q66** [comment]
it built fine previously though? is this due to the ninja switch or something

File: main/cmake/template.py [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
     "--system-zstd",
     f"--parallel={self.conf_jobs}",
 ]
-makedepends = ["libarchive-devel", "ncurses-devel", "linux-headers"]
+make_cmd = "ninja"
+makedepends = [
+    "libarchive-devel",
+    "linux-headers",
+    "ncurses-devel",
+    "ninja",
```
Comment (q66) [inline main/cmake/template.py:22]: hostmakedepends

**Comment by nekopsykose** [comment]
> is this due to the ninja switch or something  

nope, you can just uhh try a local build? just master and building current (no changes) does the same failure

**Pull request closed without merging**